### PR TITLE
Obsolete scrap weapons

### PR DESCRIPTION
**Pull request by kevingranade**
#### Summary
SUMMARY: None

#### Purpose of change
These scrap weapons have been sitting around not making any sense for years.  At the time they were added this was a bit of a different game and "weld together a bunch of pipes to make a pneumatic gun" seemed like it fit, but it doesn't anymore.

#### Describe the solution
Obsolete the crafting recipes for the scrap pneumatic guns, the mininuke launcher, and the rail rifles.

#### Describe alternatives you've considered
I'm sure someone will want to say these should go in a mod, feel free to re-add the recipes to a mod, but I'm not aware of a non-obsoleted mod in mainline that would take them.
I'm fine with having pneumatic guns, and some of those would be player craftable, but none of these are representative of any real-world pneumatic guns. I'm also not interested in working up a pneumatic gun right now.
There's a historical mininuke launcher called the Davy Crockett recoiless rifle, but it has a few problems.  It's a long range tactical weapon, it's a fairly advanced recoiless rifle design, and they don't exist any more. One of the particularly advanced factions might be able to make you a launcher, maybe as a quest lead in where you bombard something, but it's not going to even slightly resemble the thing I'm obsoleting here, so if someone is going to do that they might as well start from scratch.
The rail rifles are just a ton of nonsense, a handheld railgun is quite a stretch to start with, then using rebar for them doubles down on them not remotely resembling how a handheld railgun would work.

### CODE REVIEW

**Comment by tenmillimaster** [comment]
Shouldn't the `recipe_mininuke_launch` book and its spawns also get obsoleted?

**Comment by Night-Pryanik** [comment]
> Plus, these things existing is most likely

Most likely what?

**Comment by githubuser343** [comment]
Does "coilgun" count as a scrap weapon too? I noticed it's not included here but not sure if that was by intent.

Did not do any deep research into this but there is a whole wikipedia article https://en.wikipedia.org/wiki/Coilgun

**Comment by githubuser343** [comment]
Can't edit my post for some reason. Just want to add that the mininuke hack is also present in the military dispatch and I did not see a reference to it in the changed files.

**Comment by kevingranade** [comment]
Tl;dr "Small coilguns are recreationally made by hobbyists, typically up to several joules to tens of joules projectile energy (the latter comparable to a typical air gun and an order of magnitude less than a firearm)".

So yes, I should either remove or nerf the coil gun down to less than pistol damage.

**Comment by kevingranade** [comment]
Finished cleaning up mininuke launcher definitions.  
Obsoleted coilgun and related definitions.
Obsoleted. rail rifle and related definitions.

**Comment by Brian-Otten** [comment]
I hate seeing content go, but it was time for these to go.
I was just playing and i remembered one of the refugees in the shelter mentioning that one of their engineers had built one of the rebar rail launchers. let me see if i can find it.

**Comment by Brian-Otten** [comment]
Found it. it npc_arsonist.json

  {
    "type": "talk_topic",
    "id": "TALK_ARSONIST_DOING_REBAR",
    "dynamic_line": "Well, there's a guy downstairs who got a working pneumatic cannon.  It shoots metal like… like a cannon without the bang.  Cost-efficient as hell.  And there's no shortage of improvised weapons you can make.  The big thing though, seems to be continuing construction of fortifications.  Very few of those monsters seem to be able to break through a fence or wall constructed with the stuff.",
    "responses": [ { "text": "Well, then…", "topic": "TALK_ARSONIST" } ]
  },

Replace with (my suggestion)

  {
    "type": "talk_topic",
    "id": "TALK_ARSONIST_DOING_REBAR",
    "dynamic_line": "Well, there's a guy downstairs who got a repeating crossbow. You can shoot zombies all day with that thing and just gather your bolts afterwards.  And there's no shortage of improvised weapons you can make.  The big thing though, seems to be continuing construction of fortifications.  Very few of those monsters seem to be able to break through a fence or wall constructed with the stuff.",
    "responses": [ { "text": "Well, then…", "topic": "TALK_ARSONIST" } ]
  },

**Comment by kevingranade** [comment]
Thanks for checking, I was following the links between items and didn't think to check prose.

**Comment by kevingranade** [comment]
Oh that one, actually that's placeholder dialogue for a questline where you help them build a for real pneumatic cannon.  Like I said, it is a thing you can do, just not in the form of obvious references to other video games.

**Comment by mkikt4743** [comment]
improve, don't remove

**Comment by mkikt4743** [comment]
to be more specific, why remove these? Not only is it a fun weapon, it's a grind to get and it's fairly balanced so that you can only create it lategame. What seems to be the problem? Nothing is wrong with sci-fi in a game that has otherworldly aliens and mutagens, as well as CBMs and automated turret systems

**Comment by anothersimulacrum** [comment]
[“Seeing as we have nanobots and power armors…”, “We have teleportation, so it’s not unreasonable to have…”: :Irrelevant](https://github.com/CleverRaven/Cataclysm-DDA/wiki/Frequently-Made-Suggestions#failed-rationalizations).

**Comment by mkikt4743** [comment]
simply removing something because you don't like it is an irreverent reason to remove things.

**Comment by anothersimulacrum** [comment]
Cool, this isn't that.

File: data/mods/more_classes_scenarios/cs_classes.json [hunk 1]
```diff
@@ -66,60 +66,6 @@
       "female": [ "sports_bra", "boxer_shorts" ]
     }
   },
-  {
```
Comment (ZhilkinSerg) [inline data/mods/more_classes_scenarios/cs_classes.json:69]: Was it an intended removal?

**Comment by mkikt4743** [comment]
besides, these aren't 'scrap waepons'. saying that is misleading and puts them with say the nord or the homemade half-pike. these are complex electronic weapons

File: data/mods/more_classes_scenarios/cs_classes.json (hunk 1 above)
Comment (kevingranade) [reply data/mods/more_classes_scenarios/cs_classes.json:69]: Yes, it's in an obsoleted mod and incorporates obsoleted content, no need to keep it around.

**Comment by puyebu** [comment]
I don't think it is reasonable to remove fun things while calling it scrap.

Why not adapt it to the current game and lore and change recipes to reflect the changes rather than plainly removing things again?

**Comment by ZhilkinSerg** [comment]
> 
> 
> I don't think it is reasonable to remove fun things while calling it scrap.
> 
> Why not adapt it to the current game and lore and change recipes to reflect the changes rather than plainly removing things again?

Everything is explained in the PR description.

**Comment by Nuncio105** [comment]
Just created an account to leave a comment, as these sort of changes becomes disturbingly common.
Arbitrary removal of fun content adds nothing to the value of the game, and only pushes it towards some sort hyper realistic simulation, which is flawed concept from the beginning, given the limitations of the tile-based game with relatively small reality bubble.

In addition to biased ranting (which is no more biased than current direction of a project, i am sure more than a few people remember game being set in ~2040 and in-game mentions of cold war with china and totalitarian us government) i want to propose a solution for these particular weapons:

The game undeniably has commercially available advanced technology (police robots, commercial bionics (evidenced by in-game advertisements) which makes arguments about separation of common life-tech and military super-tech partially invalid.

Change electromagnetic guns so they use advanced electronics and capacitors salvaged from high tech weapons/equipment (laser pistols/rifles come to mind) and recipes need to be learned from lab journal (some sort of general reference for experimental electronics, which allows skilled characters to adapt some of it for more practical applications) This solution keeps distinction of common technology and super-tech (need to salvage it for irreplaceable parts), and adds more flavor and incentives to delve into labs other than mutagens.

**Comment by ZhilkinSerg** [comment]
You are on wrong premises here. Contrary to your beliefs, the primary setting of the game is present-day Earth and technology in Cataclysm is, except in specific situations, at a modern real life level.

**Comment by mkikt4743** [comment]
The PR's description only gives reasons such as 'not making sense' and 'not realistic'. No actual reason to remove this content is given other than that. Also, simply calling any argument against PRs you support irrelevant is both poor face and immature. Please give this PR an actual reason to be passed, or I'll just assume they were removed cause the devs don't like them, which again is poor face for a game.

**Comment by mkikt4743** [comment]
to quote the original PR incase you can't manage to scroll up and read it,
>it doesn't make sense anymore
>none of these are representative of real-world pneumatic guns
>a handheld railgun is quite a stretch to start with, then using rebar for them doubles down on them not remotely resembling how a handheld railgun would work.

**Comment by ZhilkinSerg** [comment]
> 
> 
> The PR's description only gives reasons such as 'not making sense' and 'not realistic'. No actual reason to remove this content is given other than that.

Yeah, description contains completely valid reasons. I don't know what else do you expect to see when something makes no sense other than "it makes no sense"?

**Comment by Soup-de-Loop** [comment]
People seem to be misunderstanding this one. While a lot of us, and I'm sure even in the dev-team find pneumatic and peculiar weaponry like coilguns etc cool, the way they exist in the game is not realistic or even representative. This PR is far more about the fact that they're in CDDA the wrong way, not necessarily that they're in it in the first place.

Just because the person removing it currently has too much on their plate to personally work on re-working them into a better integrated form doesn't mean they should still exist in their current form. What this PR is more about is opportunity for people who're low on ideas but wanting to contribute a chance to help bring this stuff into the fold.

Literally, `I'm fine with having pneumatic guns, and some of those would be player craftable, but none of these are representative of any real-world pneumatic guns`. It's not an attack on us the players or the content, it is a step to making the content better.

**Comment by mkikt4743** [comment]
eh I got a modding dude to mod them back in so it's all right now

**Comment by Mister-Cultist** [comment]
can we atleast create a catapult as a vehicle part so I can throw mininukes/large rocks?

**Comment by ZhilkinSerg** [comment]
> 
> 
> can we atleast create a catapult as a vehicle part so I can throw mininukes/large rocks?

Sure, nothing prevents you from creating anything - https://github.com/CleverRaven/Cataclysm-DDA/wiki/Guide-to-adding-new-content-to-CDDA-for-first-time-contributors

**Comment by NoahGoldFox** [comment]
This is sad. I hope somebody makes replacement weapons for these soon.

**Comment by Reango** [comment]
So for real life lethal pneumatic weaponry, I'd like to give you a reference to Joerge Sprave. https://www.youtube.com/watch?v=1U7TRDFjnFM
If the pneumatic weapons were something more like that, would that make it better? I just wanted to give you ideas.
The thing has quite a bit of power. Would easily have a long reload time.

**Comment by NoahGoldFox** [comment]
> So for real life lethal pneumatic weaponry, I'd like to give you a reference to Joerge Sprave. https://www.youtube.com/watch?v=1U7TRDFjnFM
> If the pneumatic weapons were something more like that, would that make it better? I just wanted to give you ideas.
> The thing has quite a bit of power. Would easily have a long reload time.

Yeah stuff like his would be great and fun/funny. Like most of his inventions would be liked if they were in the game.

**Comment by ta-nelson** [comment]
I think the main limitations for pneumatic rifles is you need a way to heavily compress air for them to function reasonably as a standin for say a survival rifle. For example [this one](https://www.airgundepot.com/sam-yang-dragon-claw-air-rifle.html). Even as far as a "pipe rifle" variant goes I don't think it would be an issue, the main strain on feasibility is compressing the air. Unless they are meant to run on welding gas I think it would best be reimplemented alongside an air compressor vehicle part. 

In contrast I don't think coil guns are outside the scope of the game at all. They can easily be produced at home, and the main limitation for their feasibility as actual handheld weapons is power supply, but battery technology is very sophisticated in CDDA. Maybe capacitors are the canonical bottleneck.

**Comment by FlameStormer2000** [comment]
>I'm not aware of a non-obsoleted mod in mainline that would take them.
Well gee it's almost as if you obsoleted them all. Also, why not Aftershock? Doesn't it exist specifically for deleted "unrealistic" content?

**Comment by ZhilkinSerg** [comment]
> > I'm not aware of a non-obsoleted mod in mainline that would take them.
> 
> Well gee it's almost as if you obsoleted them all. Also, why not Aftershock? Doesn't it exist specifically for deleted "unrealistic" content?

No, Aftershock exists for another reasons.

**Comment by anothersimulacrum** [comment]
Aftershock exists to be a sci-fi, not a junk heap where anything that doesn't fit in mainline goes.

**Comment by FlameStormer2000** [comment]
> > > I'm not aware of a non-obsoleted mod in mainline that would take them.
> > 
> > 
> > Well gee it's almost as if you obsoleted them all. Also, why not Aftershock? Doesn't it exist specifically for deleted "unrealistic" content?
> 
> No, Aftershock exists for another reasons.

"Aftershock, meant to restore cut content and make the game more sci-fi."

> Aftershock exists to be a sci-fi, not a junk heap where anything that doesn't fit in mainline goes.

All these weapons were entirely functional, saw use by players, and inline with the original sci-fi vision of DDA. Nothing about them was junk.

**Comment by anothersimulacrum** [comment]
It's a junk heap if it's a heap of stuff that was removed for not fitting, and that is not sci-fi.

**Comment by ZhilkinSerg** [comment]
> > > > I'm not aware of a non-obsoleted mod in mainline that would take them.
> > > 
> > > 
> > > Well gee it's almost as if you obsoleted them all. Also, why not Aftershock? Doesn't it exist specifically for deleted "unrealistic" content?
> > 
> > 
> > No, Aftershock exists for another reasons.
> 
> "Aftershock, meant to restore cut content and make the game more sci-fi."

Are you sure this citation somehow contributes to your point?

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/pneumatic-bolt-driver-rework-restoration/24492/1
